### ISSUE TITLE
New Wikidata URIs in Subject Not Saved and Label Not Displayed on Existing Works

### ISSUE THREAD

**Issue by KevinJonesMeta** [OregonDigital/OD2#2996]
### Descriptive summary

New wikidata URIs are not being saved and the label is not fetching in work form in Subject on existing works. Wikidata URIs already ingested (at migration?) on works appear to be saved but are not displaying labels or are displaying them inconsistently (haven't seen enough works to know yet).  Example URI - https://www.wikidata.org/wiki/Q7278075.

### Expected behavior

Wikidata URIs in Subject field are saved through work form and fetch label that is displayed on work after saving.

### Related work

Link to related tickets or prior related work here.

### Accessibility Concerns

Add any information here to indicate any known or suspected accessibility issues for this ticket

**Comment by wickr** [comment]
The example URI is a wikidata URI but not the 'Concept URI'. The `http` version of that URI works.

Fetch gives an error on parsing but still pulls the label:

```ruby
irb(main):314:0> sub = OregonDigital::ControlledVocabularies::Subject.new('http://www.wikidata.org/entity/Q7278075')
=> #<OregonDigital::ControlledVocabularies::Subject:0x4dbc64 ID:<http://www.wikidata.org/entity/Q7278075>>

irb(main):315:0> sub.fetch
I, [2023-12-27T01:31:04.171183 #4826]  INFO -- : Attempting to fetch http://www.wikidata.org/entity/Q7278075 from local graph cache.
[INFO] fetch_cached_graph(http://www.wikidata.org/entity/Q7278075) not found in triplestore cache (http://blazegraph.od2-prod.svc.cluster.local.:8080/bigdata/namespace/kb/sparql)
I, [2023-12-27T01:31:04.175348 #4826]  INFO -- : Fetching http://www.wikidata.org/entity/Q7278075 from the authorative source. (this is slow)
[INFO] fetch_cached_graph(http://www.wikidata.org/entity/Q7278075) not found in triplestore cache (http://blazegraph.od2-prod.svc.cluster.local.:8080/bigdata/namespace/kb/sparql)
ERROR <http://www.wikidata.org/entity/Q7278075>document/RDF/Description: nodeID addtribute '0b8bd71b926a65ca3fa72e5d9103e4d6' must be an XML Name
I, [2023-12-27T01:31:05.625724 #4826]  INFO -- : Fetched From Source
=> #<RDF::Graph:0x4dcd6c(default)>

irb(main):316:0> sub.solrize
=> ["http://www.wikidata.org/entity/Q7278075", {:label=>"RV Wecoma$http://www.wikidata.org/entity/Q7278075"}]
```

Maybe it's fine with the Concept URI - I tried creating a new work and adding a Subject and the form showed the label.

Kevin if you have other examples of the Concept URI versions not working let me know.

**Comment by KevinJonesMeta** [comment]
This is an example from 'Interviewee' field: [John Byrne Oral History Interview Recording (3 of 4) ](https://oregondigital.org/concern/videos/b2773w96w). I'll use 'http' until further notice. @wickr can you confirm that's best until there is time for this  ticket? I'll look for more examples from Subject field.

**Comment by KevinJonesMeta** [comment]
Another subject example. [Jerimiah Bonifer Oral History Transcript](https://oregondigital.org/concern/documents/k0698844j).

**Comment by KevinJonesMeta** [comment]
Correcting wikidata URIs with 'http' in 'Interviewee' field are saved and display.
Corrections to wikdiata URIs with 'http' in 'Subject' field are not saved nor display so far in [Oral Histories at Oregon State University Libraries](https://oregondigital.org/collections/osu-oral-histories) collection.

**Comment by briesenberg07** [comment]
@KevinJonesMeta has above:

>  Example URI - https://www.wikidata.org/wiki/Q7278075.

If you are retrieving Wikidata URIs from a browser, the URL in the address bar is not the entity URI. 

![image](https://github.com/user-attachments/assets/3728a959-d737-41f0-95a3-35b2f501a2d9)

Get the entity URI using the "Concept URI" link > right-click > "Copy link address" or similar:

![image](https://github.com/user-attachments/assets/6890c040-2a5e-47e4-8bed-939fdd4891a4)

Compare:

- Entity page URL: https://www.wikidata.org/wiki/Q7278075
- Entity URI: http://www.wikidata.org/entity/Q7278075

**Comment by KevinJonesMeta** [comment]
@briesenberg07 Yes, I know. changing 'wiki' to 'entity' is documented in our how-to guides and the other examples above use the concept URIs on the records. I can't remember at the moment why I provided the URL in the ticket. Probably a mistake.

**Comment by briesenberg07** [comment]
Sorry @KevinJonesMeta ! Saw your example at the top and thought it worth confirming that. Thanks for letting me know

**Comment by briesenberg07** [comment]
If you are grabbing URLs from the browser and swapping `wiki` for `entity`, but concept URIs use `http` while page URLs are `https`, was this part of the problem?

**Comment by KevinJonesMeta** [comment]
I've tried http and https in Subject field and neither is consistently working. http works in other fields thus far but I haven't tested every field yet.

**Comment by KevinJonesMeta** [comment]
Wikidata URIs that came over with migration do work.

**Comment by briesenberg07** [comment]
Very strange!

**Comment by lsat12357** [comment]
Additional QA: http wikidata uris successfully persisted with labels displaying on https://staging.oregondigital.org/concern/images/cv43nw85w. Use of https will result in the uri persisting as https but without labels; this is true of most of the vocabs, eg ons (and LOC before they were blocked in #3238). Support for https is addressed here: #1342. Only http is supported for wikidata at this time.
